### ISSUE TITLE
Variables contributed to CMIP6 archive that are *not* part of the request

### ISSUE THREAD

**Issue by taylor13** [cmip6dr/CMIP6_DataRequest_VariableDefinitions#194]
@aradhakrishnanGFDL  @senesis
We have had two queries as to how modeling groups might contribute additional variables to the CMIP6 archive that do not appear anywhere in the data request.  I will copy the email string here, so that we don't lose the discussion (from newest to oldest email)

---------------------------------------
query 1:  from Aparna at GFDL
---------------------------------------
Hi all,

For CMIP5, we were able to report additional variables (apart from what is being requested) under table names such as AmonExtras and so on. I was wondering if a similar process is approved for CMIP6 or how to go about it. 

Please advise.

Thanks,

-- 
Aparna Radhakrishnan
Engility, NOAA/GFDL

---------------------------------------
query 2:  from Stéphane at Meteo France
---------------------------------------

Dear Stéphane, dear all,

there are no technical constraints on the ESGF publisher side of things
to publish additional data. This is true if data complies to CMIP6 DRS
and files structures (file name, global attributes, and alike). Then
Martin is correct we push it to the users and might encounter the case
where groups distributes similar additional data in a way that makes the
comparison difficult. Which would sort of defeat the core purpose an
inter-comparison project.

My advice would be let's focus on data that has been requested. Let's
have a placeholder where data providers can feed additional data
(product = notRequested for example). And let's communicate on the fact
that there wont be any support on those additional data except from the
data provider.

regards
Sébastien

Le 10/05/2017 à 15:45, martin.juckes@stfc.ac.uk a écrit :
> Dear Stéphane,
>
> There are a number interpretations of "additional variables":
>   (1) New physical quantities;
>   (2) Taking an existing quantity (e.g. hfss, Surface Upward Sensible Heat Flux) and asking for it in a new configuration (e.g. different masking or frequency);
>   (2b) The new configuration could be one that exists for other variables, or it could be completely new.
>
> This approach is intended to avoid arbitrary variations in CF metadata which can occur when different people encode closely related concepts.
>
> Usability of ESGF depends on some degree of standardisation ... but I like to think of the requirements as coming from the community rather than from the software package: it would be possible to distribute less standardised data, but that just pushes work onto the users.  It is clear that data producers and data users have different priorities, even within a single institution. As you say, the WIP has a role in finding a suitable balance.
>
> On the other hand, greater flexibility in the service is likely to require greater resources ... which is beyond the scpe of the WIPs mandate,
>
> regards,
> Martin
>
> ________________________________
> From: SENESI Stéphane [stephane.senesi@meteo.fr]
> Sent: 10 May 2017 09:53
> To: Karl Taylor; balaji@princeton.edu; Juckes, Martin (STFC,RAL,RALSP); sebastien denvil
> Cc: laurent franchisteguy; marie-pierre moine; sophie valcke; SEFERIAN ROLAND; CHEVALLIER Matthieu
> Subject: Re: Publishing more variables than described in the Data Request ?
>
> Dear colleagues
>
> A few month ago, we had the discussion below on the subject of publishing additional variables, which I may summarize as :
>     - adding the variables to the DR is the recommended way; it involves getting a CF standard name; the whole is necessarily lenghty
>     - GFDL would go creating 'own' CMOR tables
>     - there was no statement regarding the behavior of the publishing software stack when presented with variables which are not known to its own version of the project's tables, nor about the possibility for it to account for 'home' tables in addition to the project's tables
>
> So, I feel that the ESGF data producers still miss some flexible mechanism (and detailed instructions) to be able to publish additional variables, even before they get a CF standard name or Martin has time to take it into account in the CMIP6 DR.
>
> Could the WIP be so kind as to elaborate some statement on this subject ?
>
> Best regards
>
> Stéphane Sénési
>
>

27 January 2017

Hi all,

On this subject, I have to stress that our workflow would better accomodate these additionnal variables to be defined in the DR : we do not use CMOR tables for generating published files.
However, the first objective is that any QC step mandatory for publishing data can be passed; hence, if these are all based on CMOR tables, we could manage with new variables being defined only there

For the time being, we have identified :
    - areacellr and sftrf, for describing the river model cell area and the fraction occupied by the river
    - radiative fluxes and other boundary layer variables (say tas, tasmin, tasmax, huss, hurs, uas, vas, was) over fresh water (lakes) and fraction of the land area grid occupied by lakes

S.
>
> ________________________________
> De: "Karl Taylor" <taylor13@llnl.gov>
> À: "martin juckes" <martin.juckes@stfc.ac.uk>, "stephane senesi" <stephane.senesi@meteo.fr>, "sebastien denvil" <sebastien.denvil@ipsl.jussieu.fr>, balaji@princeton.edu
> Cc: "laurent franchisteguy" <laurent.franchisteguy@meteo.fr>, "marie-pierre moine" <marie-pierre.moine@cerfacs.fr>, "sophie valcke" <sophie.valcke@cerfacs.fr>
> Envoyé: Mercredi 11 Janvier 2017 19:20:07
> Objet: Re: Publishing more variables than described in the Data Request ?
>
> Hi Stéphane and Martin,
>
> I agree they should be added to existing tables, but they wouldn't be
> requested for any experiment (unless the CMIP panel or some MIPs decide,
> they should be added as requested variables).
>
> best regards,
> Karl
>
> On 1/11/17 4:09 AM, martin.juckes@stfc.ac.uk wrote:
>> Hello Stéphane,
>>
>> I believe this did happen in CMIP5 as well, and the solution was to create definitions of the variables in an extra MIP table. In CMIP6 we can, in principle, add the variables into the "Extension" tables, and leave them unrequested. There should be a definition in the tables, so that QC/publishing software can check dimensions and attributes.
>>
>> As we are prioritising the finalisation of the requests from the endorsed MIPs, please try to make sure that you have CF standard names for any new variables before sending them to me.
>>
>> How many variables are you expecting to add?
>>
>> regards,
>> Martin
>>
>> ________________________________
>> From: SENESI Stéphane [stephane.senesi@meteo.fr]
>> Sent: 11 January 2017 10:47
>> To: Sébastien Denvil; V. Balaji; Karl Taylor; Juckes, Martin (STFC,RAL,RALSP)
>> Cc: FRANCHISTEGUY Laurent; marie-pierre moine; Sophie Valcke
>> Subject: Publishing more variables than described in the Data Request ?
>>
>> Hello
>>
>> It may seems odd, but we at CNRM think that we should publish some variables that are not yet considered in the Data Request. This means that there is no CMIP6 table for such variables, and hence no standard DRS filename. And that they may trigger an issue in Quality Control and at the publisher stage.
>>
>> Is anybody in a position to diagnose how the ESGF software could cope with this case ?
>>
>> Best regards
>>
>>
>> --
>> ----- Météo-France -----
>> SENESI STEPHANE

>

> ----- Météo-France -----
> SENESI STEPHANE
> CNRM/GMGEC/EST

-- 
Sébastien Denvil

**Comment by taylor13** [comment]
I agree with Martin that resources are too scarce to do much to accommodate special variables.  That being said, if resources could be made available, one option would be:

1) accept only variables that fit one of the existing tables (in terms of being consistent with the "frequency" sampling options and data structure)
2) require that any variables proposed had appropriate standard_names and that the other metadata data conformed with existing CMIP6 options.
3) require that the variable name be approved by PCMDI (or the WIP)
4) mark the data in the data request as priority = 0 (or 999?, or something else unique)
5) set the global attribute, "product", to "notRequestedOutput" or "nonStandardOutput" or "nonStandard" or something similar.

**Comment by martinjuckes** [comment]
A variable could be considered a simple addition if it is the same as an existing variable except for a change in variable name, standard name, units, long name, description and frequency. We could make a simplified form for the submission of this information and the existing variable to adapt and streamline the process.

I'd prefer to use priority 4 rather that 0, to preserve a monotonic relation between the number and the implied level of importance.

I agree with Karl's other points.

**Comment by jonseddon** [comment]
The PRIMAVERA project is generating some additional variables. Most PRIMAVERA variables are in the HighResMIP request, but we have some additional variables that are not part of HighResMIP in our own request. These are mainly existing variables at higher time resolutions or on different levels. We have generated CMOR tables for these (temporarily at: https://github.com/jonseddon/cmip6-cmor-tables). We have set the `activity_id` to `PRIMAVERA` and the table names begin with `Prim`. Everything else follows the CMIP6 standards. CEDA will be publishing this data through their ESGF node. We haven't changed the global attribute `product` currently, but can do before publication.

**Comment by taylor13** [comment]
I don't think we can accept additional tables into CMIP6, but including additional tables as part of a different "activity" is probably the easiest thing to do, as you suggest.  An ESGF search limited to CMIP and its endorsed MIPs wouldn't expose these variables, but they could be published as part of a "PRIMAVERA" project (although PrePARE wouldn't currently let them pass, I don't think).  

I wonder if the HighResMIP simulations that are also considered PRIMAVERA simulations should be identified with both activities  (which is allowed, but currently PRIMAVERA is not included in the activity_id CV).   The rules for ESGF search are here: https://docs.google.com/document/d/1jNBw2am28Hxux_YuCL_mYMi18EEGkJSGrtNntOs3PJo ) .

**Comment by mpmoine** [comment]
@jonseddon, @taylor13 

Reacting to the exchange about PRIMAVERA...

1) I am surprised reading you talking  about setting the `activity_id` to `PRIMAVERA`. I though we agreed to change the `mip_era` from` CMIP6` to `PRIMAVERA`, but keep the MIPs names in `activity_id` (i.e. `HighResMIP` for Primavera simulations).

2) about things that are not yet in the CV : 
- does Karl's answer mean that `PRIMAVERA` will be included, at the end, in CMI6 CV?
- what about `plev10` Primavera pressure axis (and maybe others) that is not "CMIP6-approved"? Is there an action to make it enter into the CMIP6 CV as well ?

**Comment by jonseddon** [comment]
@mpmoine, @taylor13  Sorry, I wasn't feeling very well on Friday and was in a rush before I went home. Yes, it's the `mip_era` that we have changed for the PRIMAVERA only data that won't be submitted to CMIP6 but will be published through ESGF. The `activity_id` was going to remain `HighResMIP` for the PRIMAVERA only data.

There shouldn't be the need for two `activity_id`'s.

Thanks @mpmoine for spotting this!

**Comment by martinjuckes** [comment]
I agree with Karl's line that we should not mix these CMOR tables (at https://github.com/PCMDI/cmor/tree/master/TestTables ) with the CMIP6 tables ... so an alternative to the temporary home provided by Jon is needed: @taylor13 : could a repository such as https://github.com/PCMDI/cmor-community be created abd uses for this, or would it be better to start a new github project?

**Comment by jonseddon** [comment]
Sorry, by temporary I meant that there is now a Met Office Github account and so they will be moved from my personal account to there. However, if a community repository were available and it would allow the PRIMAVERA data to be discovered by a wider audience then we would prefer that they go there.

**Comment by aradhakrishnanGFDL** [comment]
Hi everyone, I understand this thread has not been active in a while. But, we are seeing more of the  additional data (not in data request) publishing requests coming in at this time and I was hoping you could provide some guidance. The additional data from GFDL will also comply CF/CMOR standards.   I  want to clarify if there is a way to publish this on ESGF and get a DOI for the same since leveraging the existing workflow may be ideal. If not, we at GFDL will look into planning alternative ways to be able to publish the data and assign a DOI -- for additional data not in data request for CMIP6.

A change to mip_era was suggested on this thread, but I am not sure if its specific to PRIMAVERA.  
If examples would help, please let me know. 

Thanks for your help.
